app/controllers/root_controller.rb

```diff
@@ -0,0 +1,8 @@
+class RootController < ApplicationController
+  def index
+    @link = Link.new
+    @top_links = Link.most_popular
+    @recent_links = Link.most_recent
+    @top_users = User.top_users
```

Comment:
**Code Quality - Nitpick**
@andela-hmasila Consider using the presenter pattern to reduce the number on instance variables in the controller.
```ruby
def index
  @presenter = Links::IndexPresenter.new
end
```

In `app/presenters/links`
```ruby
module Links
  class IndexPresenter
    def top_links
      Link.most_popular
    end
    
    def recent_links
      Link.most_recent
    end
  end
end
```

Thus, in the index view you can do this:
```erb
<%= @presenter.top_links  %>
<%= @presenter.recent_links  %>
```